zad2/warriors.rb

```diff
@@ -1,12 +1,55 @@
-class Warrior
-  def initialize(name:, level:)
-  end
+class Character
+  attr_accessor :name :level
 
-  def strength
+  def initialize(name, level)
+    @name = name
+    @level = level
   end
 end
 
+class Warrior < Character
+end
+
+class Monster < Character
+end
+
 class BattleArena
   def initialize(first_character, second_character)
+    @first_character = Warrior.new
+    @second_character = Monster.new
```

Comment:
parametry podane do `initialize` nie są używane - to sprawia, że nie ma różnicy czy użyjemy `BattleArena.new(warrior, monster)` czy `BattleArena.new('anything', 'lel')` - walczyć i tak będą `Warrior` i `Monster`. `initialize` mógłby wyglądać po prostu tak:
```ruby
@first_character = first_character
@second_character = second_character
```

Wtedy o tym kto będzie walczył możemy zadecydować już przy samym użyciu klasy.